doc/specs/#3327 - Application Theming/#3327 - Application Theming.md

```diff
@@ -0,0 +1,500 @@
+`window.applicationTheme` property of these themes as well. That would likely be
+confusing, but they have that freedom.
+
+## UI/UX Design
+
+[TODO]: # TODO: We should include more mockups here. That would be nice.
+
+![Tab matches Terminal background](Tab-Matches-Terminal-Color-000.png)
+_fig 1_: Using a tab color set to "terminalBackground". The Windows PowerShell
+tab has also set its color with the color picker.
+
+![Acrylic in the titlebar](AcrylicInTitlebar.png)
+_fig 2_: Using an acrylic titlebar color, with a tab color set to
+"terminalBackground"
+
+![Titlebar matches Terminal background](AcrylicTerminalBackgroundInTitlebar.png)
+_fig 3_: Using an acrylic terminal background, and the titlebar color is set to
+"terminalBackground"
+
+
+## Capabilities
+
+### Accessibility
+
+For people using the default theming, there should not be any particular
+regressions. However, this change does open up the Terminal to changes that
+might make the Terminal less accessible with certain theme configurations. As
+these themes would all be user-defined and controlled by the user, we're not
+concerned that this will be much of an issue. If a user finds one of their
+themes is less accessible, they can always change the theme to be more
+appropriate for them, or even switch to another theme.
+
+Furthermore, this might _help_ certain accessibility stories. Users could pick
+themes with _even more_ contrast than the Terminal provides by default, or
+larger font sizes, which might help make parts of the Terminal _more_ visible
+than the default UI.
+
+### Security
+
+This should not introduce any _new_ security concerns. We're relying on the
+security of jsoncpp for parsing json. Adding new keys to the settings file
+will rely on jsoncpp's ability to securely parse those json values.
+
+### Reliability
+
+This change should not have any particular reliability concerns.
+
+### Compatibility
+
+The biggest compatibility concern is regarding the existing values for the
+`theme` property, which is addressed above.
+
+### Performance, Power, and Efficiency
+
+## Potential Issues
+
+It's totally possible for the user to set some sort of theme that just looks
+bad. This is absolutely a "beauty in the eye of the beholder" situation - not
+everyone is going to like the appearance of every theme. The goal of the
+Terminal is to provide a basic theme that's appropriate for anyone, but empower
+users to customize the terminal however they see fit. If the user chooses a
+theme that's not particularly appealing, they can always change it back.
+
+### Branding
+
+Are we concerned that by enabling theming, the appearance of the Terminal won't
+be as static, and won't necessarily have as specific a look? It might be harder
+for potential users see a screenshot of the Terminal and _know_ "Thats the
+Windows Terminal". Is this something we're really all that concerned about
+though? If this is something users want (it is), then shouldn't that be what
+matters?

```

Comment:
I think that's where the defaults matter most (I'm guessing that's a conversation we'll have at some point too)